### ISSUE TITLE
Datepicker - input field is reported as read only but is actionable on pressing enter.

### ISSUE THREAD

**Issue by aubreyquinn** [microsoft/fluentui#20979]
### Environment Information

- **Package version(s)**: "@fluentui/react-northstar": "^0.58.0"
- **Browser and OS versions**: Edge and Win 32
- **Screen Reader**: JAWS 2022.2109.9

### Describe the issue:

This issue is regarding the Datepicker component. When you supply the allowManualInput props as being false, the Datepicker puts the readonly attribute on the input field. I think that the way to fix it, would be to render a non-input field when there is no manual input. The date input field is reported as read only but is actionable on pressing enter.

### Please provide a reproduction of the issue in a codepen:
https://codesandbox.io/s/fluent-ui-example-datepicker-bug-2018938-yuuek?file=/example.js

![datepicker-bug](https://user-images.githubusercontent.com/80953505/145392240-3836a3f5-9df7-4258-8b0d-231c989e10d3.png)

#### Actual behavior:

JAWS:  input field is reported as read only but is actionable on pressing enter.

#### Expected behavior:

If they cannot edit the input field, perhaps it's better to render a button instead to open the calendar.

#### User Impact
Screen reader user will not able aware the functionality or ways by which they can edit these fields.

**Comment by gouttierre** [comment]
@aubreyquinn - Thanks for filing this issue with us. To set expectations, the developers will review this issue once capacity allows.

@jurokapsiar- Would your team be able to confirm if this is a regression, or if this behavior is an issue?
